task/git-clone-oci-ta/0.1/git_clone_oci_ta_test.go

```diff
@@ -0,0 +1,86 @@
+package git_clone_oci_ta_v01
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	tekton "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"github.com/konflux-ci/build-definitions/internal/config"
+	"github.com/konflux-ci/build-definitions/internal/kube"
+	"github.com/konflux-ci/build-definitions/internal/tkn"
+)
+
+//go:embed git-clone-oci-ta.yaml
+var taskDefinition []byte
+
+var scheme = runtime.NewScheme()
+
+func TestSimpleTest(t *testing.T) {
+	ctx := context.Background()
+
+	gitURL := "https://github.com/redhat-appstudio-qe/retrodep"
+	gitRevision := "main"
+	ociRepo := config.OCIRepo()
+	gitOCIStorage := fmt.Sprintf("%s:git-%s", ociRepo, rand.String(5))
+
+	taskSpec, err := tkn.TaskSpecFromDefinition(taskDefinition)
+	require.NoError(t, err)
+
+	taskRun := tekton.TaskRun{
+		ObjectMeta: meta.ObjectMeta{
+			GenerateName: "git-clone-oci-ta-v01-",
+			Namespace:    config.KubeNamespace(),
+		},
+		Spec: tekton.TaskRunSpec{
+			Params: []tekton.Param{
+				{Name: "url", Value: *tekton.NewStructuredValues(gitURL)},
+				{Name: "revision", Value: *tekton.NewStructuredValues(gitRevision)},
+				{Name: "ociStorage", Value: *tekton.NewStructuredValues(gitOCIStorage)},
+			},
+			TaskSpec: taskSpec,
+		},
+	}
+
+	k, err := kube.NewClient()
+	require.NoError(t, err)
+
+	require.NoError(t, k.Create(ctx, &taskRun))
+	defer k.Delete(ctx, &taskRun)
+
+	require.NoError(t, tkn.WaitForTaskRun(ctx, &taskRun, k, tkn.PollOptions{}))
+
+	require.True(t, taskRun.IsSuccessful(), fmt.Sprintf("TaskRun was not successful: %#v", taskRun.Status.Conditions))
+
+	commitResultFound := false
+	urlResultFound := false
+	sourceArtifactResultFound := false
+	for _, result := range taskRun.Status.Results {
+		switch result.Name {
+		case "commit":
+			matched, err := regexp.MatchString(`^[a-f0-9]{40}$`, result.Value.StringVal)
+			require.NoError(t, err)
+			require.True(t, matched)
+			commitResultFound = true
+		case "url":
+			require.Equal(t, result.Value.StringVal, gitURL)
+			urlResultFound = true
+		case "SOURCE_ARTIFACT":
+			expected := regexp.QuoteMeta(fmt.Sprintf("oci:%s@sha256:", ociRepo))
+			matched, err := regexp.MatchString(fmt.Sprintf("^%s[a-f0-9]{64}$", expected), result.Value.StringVal)
+			require.NoError(t, err)
+			require.True(t, matched)
+			sourceArtifactResultFound = true
+		}
+	}
+	require.True(t, commitResultFound)
+	require.True(t, urlResultFound)
+	require.True(t, sourceArtifactResultFound)
+}
```

Comment:
The tests are too verbose for my taste (though maybe that's just Go being Go)

If writing simple tests is more involved than something like this, I will not want to write them :smile: 

```
taskRun = makeTaskRun(taskSpec, params, workspaces)
run(taskRun)
assert(taskRun, expectedResults, expectedLogs)
```

I wonder if Go has something like https://github.com/samuelcolvin/dirty-equals so that you could actually just declare the expected results and wouldn't need all those `*found` booleans and switch-cases and a bunch of logic for verifying each individual result (does Gomega get close?). (And I also wonder if Go is the best choice for these tests)

Reply:
We can definitely improve on this. I'm just a bit wary of providing too many levels of indirection that will actually make it harder to read the tests.

Reply:
To be honest, for my taste, the assertions writing in Golang is not easier-to-read-and-understand than in Python. :smile:

These lines of assertions are ok to me. The assertion style choice depends on the preference of PR authors/task owners/maintenance team, etc. We can't force a unified assertion style for build-definitions, unless we want to. I think we can recommend or even force to use one assertion library in our teams and encourage custom task authors to use that.